packages/wallets/skywallet/src/adapter.ts

```diff
@@ -0,0 +1,261 @@
+
+            this._connecting = true;
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.skySolana!;
+            try {
+                await wallet.connect();
+            } catch (error: any) {
+                if (error instanceof WalletError) {
+                    throw error;
+                }
+                throw new WalletConnectionError(error?.message, error);
+            }
+
+            if (!wallet.publicKey) {
+                throw new WalletAccountError();
+            }
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+           
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection: Connection,
+        options: SendTransactionOptions = {}
+    ): Promise<TransactionSignature> {
+        let emit = true;
+        try {
+            try {
+                transaction.feePayer = transaction.feePayer || this.publicKey || undefined;
+                transaction.recentBlockhash =
+                    transaction.recentBlockhash || (await connection.getRecentBlockhash('finalized')).blockhash;
+
+                const { signers, ...sendOptions } = options;
+
+                signers?.length && transaction.partialSign(...signers);
+
+                transaction = await this.signTransaction(transaction);
+
+                const rawTransaction = transaction.serialize();
+
+                return await connection.sendRawTransaction(rawTransaction, sendOptions);
+            } catch (error: any) {
+                if (error instanceof WalletError) {
+                    emit = false;
+                    throw error;
+                }
+                throw new WalletSendTransactionError(error?.message, error);
+            }
+        } catch (error: any) {
+            if (emit) {
+                this.emit('error', error);
+            }
+            throw error;
+        }
+    }
```

Comment:
It looks like this just reimplements this entire function: https://github.com/solana-labs/wallet-adapter/blob/244af02cb3dcf9334abf61cb9a11e497bde86d57/packages/core/base/src/signer.ts#L13-L48

Why? The interface for `SkyWallet above declares a `signAndSendTransaction` method. This should be used if available.

Please don't just copy code for your adapter, make sure that it correctly adapts the interface of your wallet.